Sources/MessagingInApp/Gist/Managers/MessageQueueManager.swift

```diff
@@ -3,11 +3,16 @@ import UIKit
 
 class MessageQueueManager {
     var interval: Double = 600
-    private var queueTimer: Timer!
+    private var queueTimer: Timer?
     // The local message store is used to keep messages that can't be displayed because the route rule doesnt match.
     private var localMessageStore: [String: Message] = [:]
 
     func setup(skipQueueCheck: Bool = false) {
+        if let queueTimer {
+            queueTimer.invalidate()
+            self.queueTimer = nil
+        }
```

Comment:
Should we also add this to `deinit` of this class?

Reply:
```suggestion
            self.queueTimer?.invalidate()
            self.queueTimer = nil
```

Nit-picky suggestion. I think this code reads easier without the `if` because we are re-constructing the `queueTimer` in the line below.

Reply:
Given the instance is created and maintained from a static class, I don't think its required.